pkg/apis/batch/types.go

```diff
@@ -378,6 +378,11 @@ type CronJobSpec struct {
 	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
 	Schedule string
 
+	// The time zone for the given schedule, see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones.
+	// If not specified, this will rely on the time zone of the kube-controller-manager process.
+	// +optional
+	TimeZone string
```

Comment:
Since we are using `time.LoadLocation` for parsing, maybe we need to add some additional documentation (for users) such as:

If the name is "" or "UTC", the time zone is UTC. If the name is "Local", the time zone is local,
and it depends on the time zone of the controller manager component.

Reply:
Also, we may want to tell users that "`timeZone` field ignored when CRON_TZ or TZ used in `schedule`".
This kind of information are not supposed to be an implementation secret for users to guess.

Reply:
I can add a bit more to the documentation to it. If you're not familiar with how `time.LoadLocation` works, I could see that causing some confusion. A warning on the use of CRON_TZ or TZ is a good idea too. We already warn that it's deprecated, but not that it overrides the `timeZone` field.

Reply:
> Also, we may want to tell users that "`timeZone` field ignored when CRON_TZ or TZ used in `schedule`". This kind of information are not supposed to be an implementation secret for users to guess.

The other way around, both `CRON_TZ` and `TZ` are *not officially supported*, and you'll get warning when using them. @deejross we should ensure that if it is specified it is properly **overriden** when `timeZone` is specified.

Reply:
> We already warn that it's deprecated, but not that it overrides the `timeZone` field.

We warn it's not supported, we can extend that [warning](https://github.com/kubernetes/kubernetes/blob/3e60727f756201f63f22839523c73d7648f47948/pkg/registry/batch/cronjob/strategy.go#L121) with additional information that it will get replaced when `timeZone` is specified.

Reply:
since this is a new field, we can forbid setting both this field and CRON_TZ or TZ in schedule in validation

Reply:
I agree we should fail validation when using TZ or CRON_TZ, as it's unsupported and would be less error-prone. However, because the new field is in alpha and must be enabled via a feature gate, I thought it best to wait until it's stable and enabled by default (which the KEP points to v1.27) before removing the only other time zone workaround.

Reply:
yeah, starting by forbidding use of both the API field and an embedded TZ or CRON_TZ would be a good start that would simplify honoring of the API field